src/fusiontensor/array_cast.jl

```diff
@@ -0,0 +1,193 @@
+# This file defines interface to cast from and to generic array
+
+using BlockArrays: AbstractBlockArray, BlockedArray, blockedrange, blocklengths, findblock
+
+using BlockSparseArrays: BlockSparseArrays, BlockSparseArray
+using GradedUnitRanges: AbstractGradedUnitRange, blocklabels
+using SymmetrySectors: block_dimensions, quantum_dimension
+using TensorAlgebra: contract
+
+# =================================  High level interface  =================================
+
+#### cast from array to symmetric
+function FusionTensor(
+  array::AbstractArray,
+  codomain_legs::Tuple{Vararg{AbstractGradedUnitRange}},
+  domain_legs::Tuple{Vararg{AbstractGradedUnitRange}},
+)
+  return cast_from_array(array, codomain_legs, domain_legs)
+end
+
+#### cast from symmetric to array
+function BlockSparseArrays.BlockSparseArray(ft::FusionTensor)
+  return cast_to_array(ft)
+end
+
+# =================================  Low level interface  ==================================
+function cast_from_array(
+  array::AbstractArray,
+  codomain_legs::Tuple{Vararg{AbstractGradedUnitRange}},
+  domain_legs::Tuple{Vararg{AbstractGradedUnitRange}},
+)
+  bounds = block_dimensions.((codomain_legs..., domain_legs...))
+  blockarray = BlockedArray(array, bounds...)
+  return cast_from_array(blockarray, codomain_legs, domain_legs)
+end
+
+function cast_from_array(
+  blockarray::AbstractBlockArray,
+  codomain_legs::Tuple{Vararg{AbstractGradedUnitRange}},
+  domain_legs::Tuple{Vararg{AbstractGradedUnitRange}};
+  tol::Float64=1e-12,
```

Comment:
We should define a generic `tol` based on `eltype(blockarray)`, for example `10 * eps(eltype(blockarray))`.

Reply:
`10 * eps(real(eltype(blockarray))))` (mind the real)